flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -783,10 +783,16 @@ public CompiledPlan compilePlanSql(String stmt) {
             throw new TableException(UNSUPPORTED_QUERY_IN_COMPILE_PLAN_SQL_MSG);
         }
 
+        ModifyOperation modifyOperation = (ModifyOperation) operations.get(0);
+        // execute CREATE TABLE first for CTAS statements
+        if (modifyOperation instanceof CreateTableASOperation) {
+            CreateTableASOperation ctasOperation = (CreateTableASOperation) modifyOperation;
+            executeInternal(ctasOperation.getCreateTableOperation());
```

Comment:
After reading the related code of compile and execute plan, I think we shouldn't support ctas now because the parser only supports insert into statement, maybe we can support this feature in the future.